msteams-platform/m365-apps/using-teams-client-sdk-preview.md

```diff
@@ -0,0 +1,365 @@
+```
+
+You'll need to update the way your code calls any of these APIs to use Promises. For example, if your code is calling a Teams API like this:
+
+# [JavaScript](#tab/javascript)
+
+This code:
+
+```js
+import microsoftTeams from "@microsoft/teams-js";
+
+microsoftTeams.getContext((context) => { /* ... */ });
+```
+
+Needs to be updated to:
+
+```js
+import { app, Context } from "@microsoft/teams-js";
+
+app.getContext().then((context: Context) => {
+    /*...*/
+});
+```
+
+...or the equivalent `async/await` pattern:
+
+```js
+import { app, Context } from "@microsoft/teams-js";
+
+async function example() {
+  const context = await app.getContext();
+  /*...*/
+}
+```
+
+# [TypeScript](#tab/typescript)
+
+This code:
+
+```TypeScript
+import * as microsoftTeams from "@microsoft/teams-js";
+
+microsoftTeams.getContext((context: microsoftTeams.Context) => {
+  /* ... */
+});
+```
+
+Needs to be updated to:
+
+```TypeScript
+import { app, Context } from "@microsoft/teams-js";
+
+app.getContext().then((context: Context) => {
+    /*...*/
+});
+```
+
+...or the equivalent `async/await` pattern:
+
+```TypeScript
+import { app, Context } from "@microsoft/teams-js";
+
+async function example() {
+  const context: Context = await app.getContext();
+  /*...*/
+}
+```
+
+---
+
+> [!TIP]
+> When you update your code for the TeamsJS SDK v2 Preview using [Teams Toolkit](#updating-to-the-teams-client-sdk-v2-preview), the required updates are flagged for you with `TODO` comments in your client code.
+
+## APIs organized into capabilities
+
+A *capability* is a logical grouping of APIs that provide similar functionality. You can think of Microsoft Teams, Outlook, and Office, as hosts. A host supports a given capability if it supports all the APIs defined within that capability. A host cannot partially implement a capability.  Capabilities can be feature- or content-based, such as *dialog* or *authentication*. There are also capabilities for application types, such as *tabs/pages* or *bots*, and other groupings.
+
+In the TeamsJS SDK v2 Preview, APIs are defined as functions in a JavaScript namespace whose name matches their required capability. If an app is running in a host that supports the dialog capability, then the app can safely call APIs such as `dialog.open` (in addition to other dialog-related APIs defined in the namespace). Meanwhile, if an app attempts to call an API that's not supported in that host, the API will throw an exception.
+
+### Differentiating your app experience
+
+You can check for host support of a given capability at runtime by calling the `isSupported()` function on that capability (namespace). It will return `true` if it is supported and `false` if not, and you can adjust app behavior as appropriate. This allows your app to light up UI and functionality in hosts that support it, while continuing to run (and appear in the store) for hosts that don't.
```

Comment:
I think we want to remove "(and appear in the store)" since I don't think the ability to keep it out of the store is supported by the manifest yet.

Reply:
Ah! Good catch.